### PR TITLE
go: golint/go vet fixes

### LINKED ISSUE #1369: Golint/vet warnings in go/client/logclient_test.go
```
var fiveSeconds is of type time.Duration; don't use unit-specific suffix "Seconds"	
var sevenSeconds is of type time.Duration; don't use unit-specific suffix "Seconds"
the cancel function returned by context.WithDeadline should be called, not discarded, to avoid a context leak
```

### PR DESCRIPTION
**Pull request by daviddrysdale**
Fixes #1369

### CODE REVIEW

File: go/client/logclient_test.go [hunk 1]
```diff
@@ -229,8 +229,8 @@ func TestAddChainRetries(t *testing.T) {
 	const leeway = time.Millisecond * 100
 	const leewayRatio = 0.1 // 10%
 	const instant = time.Millisecond
-	const fiveSeconds = time.Second * 5
-	const sevenSeconds = time.Second * 7 // = 1 + 2 + 4
+	const five = time.Second * 5
+	const seven = time.Second * 7 // = 1 + 2 + 4
```
Comment (RJPercival) [inline go/client/logclient_test.go]: I'd just remove these variables and put the values directly into the test definitions below. Having them as variables doesn't seem to gain us anything. On a related note, deadlineLength and retryAfter ought to be `time.Duration`s too.

File: go/scanner/main/scanner.go [hunk 2]
```diff
@@ -18,11 +18,11 @@ import (
 )
 
 const (
-	// A regex which cannot match any input
+	// MatchesNothingRegex is a regex which cannot match any input.
 	MatchesNothingRegex = "a^"
```
Comment (RJPercival) [inline go/scanner/main/scanner.go]: Do we really need to export this?

File: go/scanner/scanner.go [hunk 3]
```diff
@@ -96,16 +106,18 @@ func (m MatchSubjectRegex) PrecertificateMatches(p *ct.Precertificate) bool {
 	return false
 }
 
-// Matches on issuer cn by regex
+// MatchIssuerRegex matches on issuer common name by regex
```
Comment (RJPercival) [inline go/scanner/scanner.go]: In some places you use the abbreviation "CN", in others you write "common name". Best to consistently use one.

Comment (RJPercival) [inline go/scanner/scanner.go]: Maybe just replace "common name" here with "CN (common name)"?

Comment (daviddrysdale) [inline go/scanner/scanner.go]: Done.

File: go/scanner/main/scanner.go (hunk 2 above)
Comment (daviddrysdale) [inline go/scanner/main/scanner.go]: Done.

File: go/client/logclient_test.go (hunk 1 above)
Comment (daviddrysdale) [inline go/client/logclient_test.go]: Removed the constants and converted `deadlineLength`; `retryAfter` isn't ever used as a `time.Duration` though.

Comment (RJPercival) [inline go/client/logclient_test.go]: `retryAfter` *is* a duration though. It's actually only used as a string, so I'd argue it's better as a `time. Duration`, with the explicitness of units that goes with that, than as an `int`.

Comment (daviddrysdale) [inline go/client/logclient_test.go]: Done, and I've unfolded the test cases to explicit named-fields.

Comment (RJPercival) [inline go/client/logclient_test.go]: Thanks, looks good!

**Pull request merged**